### PR TITLE
Feature/chioma/producer item list data page

### PR DESCRIPTION
**Pull request by Simpcyclassy**

### CODE REVIEW

File: src/components/userProfile/components/SideMenu.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { Menu, Icon } from 'antd';
 
+import UserLink from './UserLink';
```
Comment (Way2nnadi) [inline src/components/userProfile/components/SideMenu.js]: `./UserLink` does not exist

File: src/components/userProfile/components/SideMenu.js [hunk 2]
```diff
@@ -17,11 +18,15 @@ export default class SideMenu extends React.Component {
...
             >
                 {
-                    SIDE_MENU_ITEMS.map(sideMenuItem => (
-                        <Menu.Item key={sideMenuItem.label}>
-                            <span><Icon type={sideMenuItem.iconType} />{sideMenuItem.label}</span>
+                    SIDE_MENU_ITEMS.map(({ path, iconType, label }) => (
+                        <Menu.Item key={path}>
+                            <span>
+                                <Icon type={iconType} />
+                                <UserLink key={path} to={path}>{label}</UserLink>
```
Comment (Way2nnadi) [inline src/components/userProfile/components/SideMenu.js]: What are you trying to do here?

File: src/components/userProfile/constants.js [hunk 3]
```diff
@@ -70,11 +75,176 @@ export const LOCATION = [
...
-    { iconType: "notification", label: "Notifications" },
-    { iconType: "shopping", label: "Purchases" },
-    { iconType: "folder", label: "Saved Items" },
-    { iconType: "setting", label: "Settings" },
-    { iconType: "solution", label: "Help" },
+    { iconType: "info-circle", label: "Account Info", path:"./components/ProfileContent" },
+    { iconType: "message", label: "Messages", path:"/" },
+    { iconType: "notification", label: "Notifications", path:"/" },
+    { iconType: "shopping", label: "Purchases", path:"/" },
+    { iconType: "folder", label: "Add Items", path:"./components/AddItem" },
```
Comment (Way2nnadi) [inline src/components/userProfile/constants.js]: `./components/AddItem` is a component not a route path

File: src/components/userProfile/components/SideMenu.js (hunk 2 above)
Comment (Simpcyclassy) [reply src/components/userProfile/components/SideMenu.js]: I intended to use it to access the paths

File: src/components/producerData/components/AddItem.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { List, Avatar, Icon } from 'antd';
+import { LIST_DATA, PRODUCER_PAGE } from '../constants';
+
+const { EDIT, LARGE, LOGO, PRODUCT_ITEM, VERTICAL } = PRODUCER_PAGE;
+
+class AddItem extends React.Component {
+    render() {
+        return (
+            <div className={PRODUCT_ITEM}>
```
Comment (enyeInc) [inline src/components/producerData/components/AddItem.js]: unneeded `div` move className to `List` component

File: src/components/app/constants.js [hunk 5]
```diff
@@ -28,12 +28,14 @@ export const PATHS = {
...
     HOME: '/',
     MARKET: '/market',
+    PRODUCER_DATA_PAGE: '/producerData',
     PROFILE: '/profile',
     SIGN_IN: '/signin',
     SIGN_UP: '/signup',
 };
 
 export const USER_AVATAR = {
+    ADD_ITEM: "Add Item",
```
Comment (enyeInc) [inline src/components/app/constants.js]: this variable is not part of `USER_AVATAR` move it out of this

File: src/components/userProfile/components/SideMenu.js [hunk 6]
```diff
@@ -17,11 +17,12 @@ export default class SideMenu extends React.Component {
...
-                    SIDE_MENU_ITEMS.map(sideMenuItem => (
-                        <Menu.Item key={sideMenuItem.label}>
-                            <span><Icon type={sideMenuItem.iconType} />{sideMenuItem.label}</span>
+                    SIDE_MENU_ITEMS.map(({ path, iconType, label }) => (
+                        <Menu.Item key={path}>
+                            <span><Icon type={iconType} />{label}</span>
                         </Menu.Item>
-                    ))
+                    )
+                    )
```
Comment (enyeInc) [inline src/components/userProfile/components/SideMenu.js]: `))` is cleaner

File: src/components/userProfile/components/UserLink.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+// UserLink.propTypes = {
+//     children: PropTypes.oneOfType([
+//         PropTypes.arrayOf(PropTypes.node),
+//         PropTypes.node,
+//     ]),
+//     to: PropTypes.string.isRequired,
+// };
+
+// export default UserLink;
```
Comment (enyeInc) [inline src/components/userProfile/components/UserLink.js]: delete this file if you are not using it

File: src/components/producerData/components/AddItem.js (hunk 4 above)
Comment (Simpcyclassy) [reply src/components/producerData/components/AddItem.js]: OK

File: src/components/app/constants.js (hunk 5 above)
Comment (Simpcyclassy) [reply src/components/app/constants.js]: OK

Comment (enyeInc) [reply src/components/app/constants.js]: This variable hasn't been moved

**Comment by enyeInc** [comment]
@Simpcyclassy looks good to me, you just have to fix the `ADD_ITEM` variable in `USER_AVATAR` object

File: src/components/userProfile/components/SideMenu.js [hunk 8]
```diff
@@ -17,9 +17,9 @@ export default class SideMenu extends React.Component {
                 mode={INLINE}
             >
                 {
-                    SIDE_MENU_ITEMS.map(sideMenuItem => (
-                        <Menu.Item key={sideMenuItem.label}>
-                            <span><Icon type={sideMenuItem.iconType} />{sideMenuItem.label}</span>
+                    SIDE_MENU_ITEMS.map(({ path, iconType, label }) => (
+                        <Menu.Item key={path}>
+                            <span><Icon type={iconType} />{label}</span>
```
Comment (joshuaobinna) [inline src/components/userProfile/components/SideMenu.js]: <span><Icon type={iconType} />{label}</span>
iconType should be capitalized if its a constant. 
if it is edit call it EDIT

**Comment by joshuaobinna** [comment]
looks good to me

File: src/components/app/constants.js (hunk 5 above)
Comment (Simpcyclassy) [reply src/components/app/constants.js]: Hmmm. I did move it. Something is definitely wrong somewhere

File: src/components/userProfile/components/SideMenu.js [hunk 9]
```diff
@@ -17,9 +17,9 @@ export default class SideMenu extends React.Component {
                 mode={INLINE}
             >
                 {
-                    SIDE_MENU_ITEMS.map(sideMenuItem => (
-                        <Menu.Item key={sideMenuItem.label}>
-                            <span><Icon type={sideMenuItem.iconType} />{sideMenuItem.label}</span>
+                    SIDE_MENU_ITEMS.map(({ path, iconType, label }) => (
```
Comment (Way2nnadi) [inline src/components/userProfile/components/SideMenu.js]: the objects in `SIDE_MENU_ITEMS` do not contain a `path` key.  remove it from this piece of logic and use `label` as the `key` for the `Menu.Item`s

Comment (Simpcyclassy) [reply src/components/userProfile/components/SideMenu.js]: OK

**Comment by Way2nnadi** [comment]
@Simpcyclassy looks good to me

File: package.json [hunk 10]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "agroeasy",
-  "version": "1.1.5",
+  "version": "1.2.0",
```
Comment (EngrEric) [inline package.json]: why did you change the minor version?

**Comment by EngrEric** [comment]
looks good but you need to resolve the conflict in the package.json file.
And your branch is behind the latest master

File: package.json (hunk 10 above)
Comment (EngrEric) [reply package.json]: cool,

**Comment by Way2nnadi** [comment]
@EngrEric I am not getting the same issue as you are getting. Try pulling her most recent code

File: package.json (hunk 10 above)
Comment (Simpcyclassy) [reply package.json]: Alright

File: src/components/router/components/Routes.js [hunk 11]
```diff
@@ -17,6 +18,12 @@ const Routes = () => (
                 <Route path={HOME} exact strict component={home.components.Home} />
                 <Route path={MARKET} exact strict component={home.components.Home} />
                 <Route path={PROFILE} exact strict component={userProfile.components.UserProfile} />
```
[the remark this answers is not in the document]
Comment (Simpcyclassy) [reply src/components/router/components/Routes.js]: OK

**Comment by EngrEric** [comment]
> @EngrEric I am not getting the same issue as you are getting. Try pulling her most recent code

ok, its actually working well.
my local branch of her branch is not updated then.

File: package.json (hunk 10 above)
Comment (joshuaobinna) [reply package.json]: Well said

**Comment by joshuaobinna** [comment]
Good job chioma
@EngrEric are the card components in my branch aligned at ur end?

**Comment by EngrEric** [comment]
> Good job chioma
> @EngrEric are the card components in my branch aligned at ur end?

yes

**Comment by Way2nnadi** [comment]
@joshuaobinna Keep question on this PR about Chioma's code given that we are peer reviewing her code here
